File: src/main/org/nlogo/prim/_patch.java
package org.nlogo.prim;

import org.nlogo.api.LogoException;
import org.nlogo.api.Nobody$;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _patch
    extends Reporter {
  @Override
  public Syntax syntax() {
    int[] right;
    right = new int[]{Syntax.TYPE_NUMBER,
        Syntax.TYPE_NUMBER};

    return Syntax.reporterSyntax
        (right, Syntax.TYPE_PATCH | Syntax.TYPE_NOBODY);
  }

  // I've tried to rejigger this and the result gets past TryCatchSafeChecker but then
  // doesn't work at runtime ("Inconsistent stack height") - ST 2/10/09
  @Override
  public Object report(final Context context) throws LogoException {
    try {
      return
          world.getPatchAt
              (argEvalDoubleValue(context, 0),
                  argEvalDoubleValue(context, 1));
    } catch (org.nlogo.api.AgentException ex) {
      return Nobody$.MODULE$;
    }
  }
}

File: netlogo-headless/src/main/prim/etc/_patch.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.AgentException
import org.nlogo.core.Nobody
import org.nlogo.nvm.{ Context, Reporter }

class _patch extends Reporter {

  // I've tried to rejigger this and the result gets past TryCatchSafeChecker but then doesn't work
  // at runtime ("Inconsistent stack height") - ST 2/10/09
  override def report(context: Context): AnyRef =
    try world.getPatchAt(
      argEvalDoubleValue(context, 0),
      argEvalDoubleValue(context, 1))
    catch {
      case _: AgentException => Nobody
    }

}
